### PR TITLE
Issue #3 - Create a fragment for pet details and a pet details activity that hosts it

### PR DESCRIPTION
**Pull request by smacgregor**
https://github.com/TeamCritterFinder/CritterFinder/issues/3 - Scaffolding to support a pet details fragment which can be re-used throughout the app.

This PR is dirty and depends on https://github.com/TeamCritterFinder/CritterFinder/pull/2 and https://github.com/TeamCritterFinder/CritterFinder/issues/3.
1. Create a Pet Details Fragment and an associated fragment layout
2. Create a Pet Details Activity which hosts the fragment. We may not end up using this activity - but for now I need it as a way to test the fragment.
3. Create subpackages in the project for fragments, activities and adapters

Screen shot of the changes:

screen shot 2016-02-29 at 11 20 48 pm

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -2,21 +2,20 @@
...
-        android:minSdkVersion="16"
-        android:targetSdkVersion="21" />
+    android:versionName="1.0">
 
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
 
+    <uses-permission android:name="android.permission.READ_PHONE_STATE" />
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+
```
Comment (smacgregor) [inline app/src/main/AndroidManifest.xml:12]: auto generated by android studio when the SDK was updated to target version 23.

File: app/src/main/java/com/codepath/apps/critterfinder/activities/PetDetailsActivity.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+     * @param context
+     * @param pet
+     * @return
+     */
+    public static Intent getStartIntent(Context context, PetModel pet) {
+        Intent intent= new Intent(context, PetDetailsActivity.class);
+        // TODO - once our pet model is parcelabe, work our magic
+        intent.putExtra(PetDetailsActivity.EXTRA_PET, "");
+        return intent;
+    }
```
Comment (smacgregor) [inline app/src/main/java/com/codepath/apps/critterfinder/activities/PetDetailsActivity.java:30]: static function to allow consumers to get an intent that will launch this activity.

File: app/src/main/java/com/codepath/apps/critterfinder/activities/PetDetailsActivity.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            // TODO - pass in the pet
+            setupPetDetails(null);
+        }
+    }
+
+    private void setupPetDetails(PetModel pet) {
+        PetDetailsFragment petDetailsFragment = PetDetailsFragment.newInstance(null);
+        getSupportFragmentManager().beginTransaction().
+                replace(R.id.layout_details_fragment_placeholder, petDetailsFragment).
+                commit();
```
Comment (smacgregor) [inline app/src/main/java/com/codepath/apps/critterfinder/activities/PetDetailsActivity.java:51]: inject the fragment to prove that it works

File: app/src/main/java/com/codepath/apps/critterfinder/fragments/PetDetailsFragment.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    public static PetDetailsFragment newInstance(PetModel pet) {
+        PetDetailsFragment petDetailsFragment = new PetDetailsFragment();
+        Bundle args = new Bundle();
+
+        // TODO - once our pet model is parcelable wire it up
+        args.putString(ARGUMENT_PET, "");
+        petDetailsFragment.setArguments(args);
+        return petDetailsFragment;
+    }
```
Comment (smacgregor) [inline app/src/main/java/com/codepath/apps/critterfinder/fragments/PetDetailsFragment.java:33]: static function that creates an instance of our pet details fragment

File: app/src/main/java/com/codepath/apps/critterfinder/fragments/PetDetailsFragment.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_pet_details, container, false);
+        ButterKnife.bind(this, view);
+
+        setupPetDetailsView();
+        return view;
+    }
+
+    private void setupPetDetailsView() {
+        mPetDetailsLabel.setText(getResources().getString(R.string.pet_details_description));
```
Comment (smacgregor) [inline app/src/main/java/com/codepath/apps/critterfinder/fragments/PetDetailsFragment.java:45]: place holder to prove the fragment is getting created correctly

File: app/src/main/res/layout/activity_pet_details.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (smacgregor) [inline app/src/main/res/layout/activity_pet_details.xml:1]: auto generated blank activity for pet details

File: app/src/main/res/layout/content_pet_details.xml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    android:paddingTop="@dimen/activity_vertical_margin"
+    app:layout_behavior="@string/appbar_scrolling_view_behavior"
+    tools:context="com.codepath.apps.critterfinder.activities.PetDetailsActivity"
+    tools:showIn="@layout/activity_pet_details">
+
+    <FrameLayout
+        android:id="@+id/layout_details_fragment_placeholder"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+    </FrameLayout>
```
Comment (smacgregor) [inline app/src/main/res/layout/content_pet_details.xml:19]: we'll use this frame layout as our place holder. The activity replaces this with an instance of our pet details fragment at run time.

File: app/src/main/res/layout/fragment_pet_details.xml [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:orientation="vertical"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <TextView
+        android:id="@+id/text_pet_details"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content" />
```
Comment (smacgregor) [inline app/src/main/res/layout/fragment_pet_details.xml:12]: a simple text view for testing.

File: app/src/main/res/values-v21/styles.xml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<resources>>
+
+    <style name="AppTheme.NoActionBar">
+        <item name="windowActionBar">false</item>
+        <item name="windowNoTitle">true</item>
+        <item name="android:windowDrawsSystemBarBackgrounds">true</item>
+        <item name="android:statusBarColor">@android:color/transparent</item>
+    </style>
+</resources>
```
Comment (smacgregor) [inline app/src/main/res/values-v21/styles.xml:9]: auto generated by android studio. I didn't create this.

File: app/src/main/res/values-w820dp/dimens.xml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+<resources>
+    <!-- Example customization of dimensions originally defined in res/values/dimens.xml
+         (such as screen margins) for screens with more than 820dp of available width. This
+         would include 7" and 10" devices in landscape (~960dp and ~1280dp respectively). -->
+    <dimen name="activity_horizontal_margin">64dp</dimen>
+</resources>
```
Comment (smacgregor) [inline app/src/main/res/values-w820dp/dimens.xml:6]: auto generated by android studio. I didn't create this.

**Comment by scottrichards** [comment]
I verified this locally and merged locally

**Pull request merged**